File: server/src/main/java/se/kth/id2203/core/DataMessage.scala
package se.kth.id2203.core

import se.kth.id2203.core.ExercisePrimitives._
import se.sics.kompics.KompicsEvent
;

/**
  * Created by sindrikaldal on 17/02/17.
  */
case class DataMessage(timestamp: VectorClock, payload: KompicsEvent) extends KompicsEvent;

File: server/src/main/java/se/kth/id2203/broadcasting/DataMessage.java
package se.kth.id2203.broadcasting;

import se.sics.kompics.KompicsEvent;

import java.io.Serializable;

/**
 * Created by sindrikaldal on 23/02/17.
 */
public class DataMessage implements KompicsEvent, Serializable {

    public final VectorClock vec;
    public final KompicsEvent payload;

    public DataMessage(VectorClock vec, KompicsEvent payload) {
        this.vec = vec;
        this.payload = payload;
    }
}
